api/src/test/java/jakarta/data/metamodel/impl/AttributeRecordTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel.impl;
+
+import jakarta.data.metamodel.Attribute;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class AttributeRecordTest {
+
+    interface _SimpleEntity {
+        String ID = "id";
+        String DESCRIPTION = "description";
+
+        Attribute<SimpleEntity> id = new AttributeRecord<>(ID);
+        Attribute<SimpleEntity> description = new AttributeRecord<>(DESCRIPTION);
+    }
+
+    // A simple test entity
+    static class SimpleEntity {
+        String id;
+        String description;
+    }
+
+    @Test
+    @DisplayName("should create attribute record with name and validate interface type")
+    void shouldCreateAttributeRecordWithName() {
+        var attribute = new AttributeRecord<>("id");
+
+        SoftAssertions.assertSoftly(soft -> {
+            soft.assertThat(attribute.name()).isEqualTo("id");
+            soft.assertThat(attribute).isInstanceOf(Attribute.class);
+            soft.assertThat(attribute.name()).isNotBlank();
```

Comment:
A few of these assertions are somewhat amusing, like checking that a value is not blank after having already checked that it was equal to a particular non-blank String value 2 lines above.  But everything looks fine. Approving.

Reply:
Ok, I will remove those and merge it.